solr/core/src/java/org/apache/solr/update/TransactionLog.java

```diff
@@ -238,9 +240,6 @@ protected TransactionLog(
       }
 
       success = true;
-
-      assert ObjectReleaseTracker.track(this);
```

Comment:
In addition to the scope of the PR, I did a little move of this line to the bottom where it's supposed to be.